code/modules/mob/living/simple_animal/hostile/megafauna/ancient_robot.dm

```diff
@@ -0,0 +1,748 @@
+	. = ..()
+	TR = new /mob/living/simple_animal/hostile/ancient_robot_leg(loc, src, TOP_RIGHT)
+	TL = new /mob/living/simple_animal/hostile/ancient_robot_leg(loc, src, TOP_LEFT)
+	BR = new /mob/living/simple_animal/hostile/ancient_robot_leg(loc, src, BOTTOM_RIGHT)
+	BL = new /mob/living/simple_animal/hostile/ancient_robot_leg(loc, src, BOTTOM_LEFT)
+	beam = new /obj/effect/abstract(loc)
+	addtimer(CALLBACK(src, .proc/leg_setup), 1 SECONDS)
+	mode = pick(BLUESPACE, GRAV, PYRO, FLUX, VORTEX) //picks one of the 5 cores.
+	if(mode == FLUX) // Main attack is shock, so flux makes it stronger
+		melee_damage_lower = 25
+		melee_damage_upper = 25
+	body_shield()
+	add_overlay("[mode]")
+	add_overlay("eyes")
+
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/Destroy()
+	QDEL_NULL(TR)
+	QDEL_NULL(TL)
+	QDEL_NULL(BR)
+	QDEL_NULL(BL)
+	QDEL_NULL(beam)
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/proc/leg_setup()
+	fix_specific_leg(TOP_RIGHT)
+	fix_specific_leg(TOP_LEFT)
+	fix_specific_leg(BOTTOM_RIGHT)
+	fix_specific_leg(BOTTOM_LEFT)
+
+/obj/item/gps/internal/ancient
+	icon_state = null
+	gpstag = "Malfunctioning Signal"
+	desc = "ERROR_NULL_ENTRY"
+	invisibility = 100
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/death(gibbed, allowed = FALSE)
+	if(allowed)
+		return ..()
+	else if(exploding) //but it refused
+		return
+	adjustBruteLoss(-1)
+	self_destruct()
+	exploding = TRUE
+
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/Life(seconds, times_fired)
+	..()
+	if(!exploding)
+		return
+	playsound(src, 'sound/items/timer.ogg', 70, 0)
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/drop_loot()
+	var/core_type = null
+	switch(mode)
+		if(BLUESPACE)
+			core_type = /obj/item/assembly/signaler/anomaly/bluespace
+		if(GRAV)
+			core_type = /obj/item/assembly/signaler/anomaly/grav
+		if(PYRO)
+			core_type = /obj/item/assembly/signaler/anomaly/pyro
+		if(FLUX)
+			core_type = /obj/item/assembly/signaler/anomaly/flux
+		if(VORTEX)
+			core_type = /obj/item/assembly/signaler/anomaly/vortex
+
+	var/crate_type = pick(loot)
+	var/obj/structure/closet/crate/C = new crate_type(loc)
+	new core_type(C)
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/OpenFire()
+	if(charging)
+		return
+
+	if(exploding)
+		return
+
+	anger_modifier = clamp(((maxHealth - health)/50),0,20)
+	ranged_cooldown = world.time + (ranged_cooldown_time * ((10 - extra_player_anger)/10))
+
+	if(prob(30 + anger_modifier))
+		triple_charge()
+
+	else if(prob(15 + anger_modifier))
+		spawn_anomalies()
+
+	else if(prob(60 + anger_modifier))
+		do_special_move()
+
+	calculate_extra_player_anger()
+
+/mob/living/simple_animal/hostile/megafauna/ancient_robot/proc/triple_charge()
+	if(mode == BLUESPACE)
+		charge(delay = 24) //An extra charge, to make up for the longer time between teleports
+		charge(delay = 18)
+		charge(delay = 12)
+		charge(delay = 6)
+		SetRecoveryTime(15)
```

Comment:
the `SetRecoveryTime` doesnt need to be in the `if => else` as it happens in both. just put it at the end of the proc.